lib/datasource/maven.js

```diff
@@ -0,0 +1,123 @@
+const _ = require('lodash');
+const got = require('got');
+const url = require('url');
+const fs = require('fs-extra');
+const xmlParser = require('fast-xml-parser');
+
+module.exports = {
+  getPkgReleases,
+  resetCache,
+};
+
+let map = new Map();
+
+function resetCache() {
+  map = new Map();
+}
+
+// eslint-disable-next-line no-unused-vars
+async function getPkgReleases(purl, config) {
+  const versions = [];
+  const dependency = getDependencyParts(purl);
+  const repositories = getRepositories(purl);
+  logger.debug(
+    `Found ${repositories.length} repositories for ${dependency.display}`
+  );
+  for (let i = 0; i < repositories.length; i += 1) {
+    const repoUrl = repositories[i];
+    logger.debug(
+      `Looking up ${dependency.display} in repository #${i} - ${repoUrl}`
+    );
+    const mavenMetadata = await downloadMavenMetadata(dependency, repoUrl);
+    if (mavenMetadata) {
```

Comment:
Should we return early if we find the dependency in a repository? Or is there a reason to keep querying other repositories even after we have a result? And if we *don't* return early then we probably need a function to de-duplicate results/versions from different repositories.

Reply:
It is possible that you find versions in several repos so we need to make sure we get all of them. The deduplication is right underneath

Reply:
Is it a common case that you find different versions in several repos, or an uncommon use case? What causes it to happen that way? I assume we need to keep it as you describe, but I'd like to understand it.

Reply:
It's uncommon, but I found that case a couple of times myself. Sometimes projects migrate from repositories and they leave the previous versions in the old repo and start updating only the new repo. Happens more often with old projects:
https://search.maven.org/search?q=g:jboss%20AND%20a:jboss&core=gav
https://bintray.com/bintray/jcenter/jboss%3Ajboss
For some reason in this dependency jcenter have a newer version than maven :?

gradle will look for every repository in order looking for the specific dependency